libevmasm/KnownState.h

```diff
@@ -100,10 +100,12 @@ class KnownState
 	void resetStorage() { m_storageContent.clear(); }
 	/// Resets any knowledge about storage.
 	void resetMemory() { m_memoryContent.clear(); }
+	/// Resets known Keccak-256 hashes
+	void resetKnownKeccak256Hashes() { m_knownKeccak256Hashes.clear(); }
```

Comment:
Added a new function to clear the `m_knownKeccak256Hashes` whenever memory is reset. Forcefully reset makes the code 'look' safer, even though I think it makes no difference.

Reply:
I don't think this is needed: `loadFromMemory` returns "unknown" after a memory clear and the sequence number is correctly incremented.

Reply:
I think it's not needed, either. But doesn't it make sense to keep it for consistency with the rest of the resets? I also think it feels safer to reset the keccak hashes when memory is reset.